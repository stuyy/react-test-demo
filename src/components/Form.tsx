import React, { useState } from 'react';
import { submitForm } from '../utils/api';

export const Form = () => {
  const [username, setUsername] = useState<string>('');
  const [password, setPassword] = useState<string>('');
  const [error, setError] = useState<string>('');

  const onSubmit = async (e: React.FormEvent<HTMLFormElement>) => {
    e.preventDefault();
    try {
      const response = await submitForm({ username, password });
      setError('');
    } catch (err) {
      setError('Something went wrong!');
    }
  };

  return (
    <form onSubmit={onSubmit}>
      <div>
        <h1>{error}</h1>
        <label htmlFor="username">Username</label>
        <input
          id="username"
          value={username}
          onChange={(e) => setUsername(e.target.value)}
        />
      </div>
      <div>
        <label htmlFor="password">Password</label>
        <input
          id="password"
          value={password}
          onChange={(e) => setPassword(e.target.value)}
        />
      </div>
      <button type="submit">Submit</button>
    </form>
  );
};
