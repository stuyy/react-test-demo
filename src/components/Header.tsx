import React from 'react';

export const Header = () => (
  <header
    style={{
      width: '100%',
      height: '100px',
      display: 'flex',
      justifyContent: 'space-between',
      alignItems: 'center',
      backgroundColor: '#272727',
      color: 'white',
      padding: '20px 35px',
      boxSizing: 'border-box',
    }}
  >
    <div>
      <h1>Testing App Demo</h1>
    </div>
    <div>
      <button>Login</button>
    </div>
  </header>
);
