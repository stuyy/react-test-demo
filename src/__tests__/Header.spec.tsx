import { render } from '@testing-library/react';
import { Header } from '../components/Header';

describe('Header', () => {
  it('should match snapshot', () => {
    const component = render(<Header />);
    expect(component).toMatchSnapshot();
  });
  it('should have Testing App Demo text', () => {
    const component = render(<Header />);
    const titleText = component.getByText('Testing App Demo');
    expect(titleText).toBeInTheDocument();
    expect(titleText).toBeInstanceOf(HTMLHeadingElement);
  });
  it('should have Login button', () => {
    const component = render(<Header />);
    const btn = component.getByRole('button', { name: 'Login' });
    expect(btn).toBeInTheDocument();
  });
});
