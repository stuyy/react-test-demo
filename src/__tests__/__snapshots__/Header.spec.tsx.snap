// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header
        style="width: 100%; height: 100px; display: flex; justify-content: space-between; align-items: center; background-color: rgb(39, 39, 39); color: white; padding: 20px 35px; box-sizing: border-box;"
      >
        <div>
          <h1>
            Testing App Demo
          </h1>
        </div>
        <div>
          <button>
            Login
          </button>
        </div>
      </header>
    </div>
  </body>,
  "container": <div>
    <header
      style="width: 100%; height: 100px; display: flex; justify-content: space-between; align-items: center; background-color: rgb(39, 39, 39); color: white; padding: 20px 35px; box-sizing: border-box;"
    >
      <div>
        <h1>
          Testing App Demo
        </h1>
      </div>
      <div>
        <button>
          Login
        </button>
      </div>
    </header>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
